Add fill_value option for grids failed to interpolate in the fit

lan496 opened the pull request:
Fixes https://github.com/lmhale99/atomman/issues/24

Sorry in advance that there are lines not related to this fix because some trailing blanks are automatically deleted by my editor.

lan496 committed:
Add warnings when failed to interpolate

lmhale99 commented:
This is a good idea to add!

Maybe the fill_value parameter should also be made accessible by adding it to the interpolate_contour() and grid_interpolate_2d() calls?  I could see this being an alternate choice for property values that are constant in bulk. The only downside is that the warning would then be ignored as it only checks for nans. Then again, if people set fill_value it would indicate they know about the problem.

lan496 committed:
Add fill_value option in interpolateion

`interpolate_contour` may return nan when failed to interpolate grid
array in the fit. This fix adds `fill_value` option to make it
accessible to fill in the failed points with a given value. Also, when
extrapolation in the fit occures with the default `fill_value`, a
warning message to suggest to use `fill_value` is displayed.

lan496 commented:
Thanks for your reply.

> Maybe the fill_value parameter should also be made accessible by adding it to the interpolate_contour() and grid_interpolate_2d() calls?

I totally agree! Following your comment, I have added `fill_value` option for `interpolate_contour`. When it is set other than `np.nan`, the warning for extrapolation is disabled.
